Design-README

maosakana opened the pull request:
#  what
change README.md


# why
because it becomes necessary for development

maosakana committed:
change-README

-databese

kawa18sima commented:
まずプルリク出す際は説明を入れてください
<img width="700" alt="2017-07-09 13 03 50" src="https://user-images.githubusercontent.com/24731933/27991107-1a862478-64a7-11e7-94ce-7fa06cd4eec3.png">
こんな感じで

また修正したらプルリクを送ってください

kawa18sima commented:
プルリク出す際は説明を入れてください
<img width="700" alt="2017-07-09 13 03 50" src="https://user-images.githubusercontent.com/24731933/27991107-1a862478-64a7-11e7-94ce-7fa06cd4eec3.png">
こういう感じでお願いします

maokozuma commented:
## what
change README.md
Database Design
## why
because it becomes necessary for development

maosakana committed:
change : to -

# whtat
change : to -

# why
It makes easy to check
To have a sense of unity

maosakana commented on README.md:
```diff
@@ -1,24 +1,49 @@
+## groups
+**association**
+- has_many messages
+- has_many group_users
```
修正しました！

maosakana committed:
change place of the association

# why
it makes easy to check
and to have a sense of unity

maosakana commented on README.md:
```diff
@@ -1,24 +1,49 @@
+|user_id|integer|null: false, foreign_key: true|
+|group_id|integer|null: false, foreign_key: true|
+
+**association**
```
修正しました！

maosakana committed:
change "users" to "user"

# why
because “belongs_to” is written in singular

maosakana commented on README.md:
```diff
@@ -1,24 +1,49 @@
+|email                |string |null: false|
+|password             |string |nill: false|
+
+## membersテーブル
```
messageとmemberが混ざって混乱してました🙇💦
確認してからプルリク送るようにします！

maosakana replied:
修正しました！

maosakana committed:
add message table

# what
add column body and image

# why
because it is necessary of database

maosakana committed:
delete "id" column

# why

be created id column on default

maosakana committed:
delete columns

# what
delete columns; email and password

# why
because there are their columns on devise gem on default

kawa18sima commented on README.md:
```diff
@@ -1,24 +1,46 @@
-* Services (job queues, cache servers, search engines, etc.)
+|column  |type      |
+|--------|----------|
+|user_id |references|
```
外部キーを設定するならforeign_keyを忘れないで下さい

maosakana replied:
修正しました。！

kawa18sima commented on README.md:
```diff
@@ -1,24 +1,46 @@
+|column  |type      |
+|--------|----------|
+|user_id |references|
+|group_id|references|
```
ここも

maosakana replied:
修正しました！

kawa18sima commented on README.md:
```diff
@@ -1,24 +1,46 @@
-* Configuration
 
-* Database creation
+## groups
```
userテーブルとgroupテーブルのリレーションについて教えてもらえますか？

maosakana replied:
（修正した後のリレーション）
userとgroupは多対多の関係にあるので，group_userを結合モデルに起きました。
（has_many through）

kawa18sima commented on README.md:
```diff
@@ -1,24 +1,46 @@
+|name                 |string |null: false |
 
-* Ruby version
+## message
```
複数形で書いてください

maosakana replied:
修正しました！

kawa18sima commented:
全部同じブランチでやって大丈夫です。

コミットについてですが、少し細かすぎるので１ファイル１コミットでも大丈夫です。
ただし、１ファイルにたくさんの機能を含ませる場合はその都度コミットで切ったほうが管理しやすくなります。

maosakana committed:
fix-README

maosakana commented on README.md:
```diff
@@ -27,7 +27,7 @@
 **association**
 - has_many messages
 - has_many group_users
```
## what
fix 4 parts
- change "message" to "messages" for double 
- fix type of user_id and group_id  in messages table
   change "integer" to "references"  
- fix groups relation about user to "has_many through"
- add options to group_user table

## why
- Because it is required to deverpment

kawa18sima commented:
いいんじゃないでしょうか。
マージして次に進みましょう!!!!

maokozuma commented:
ありがとうございます！！！！！

maosakana committed:
create-database
